### PR TITLE
Implement waiters

### PR DESCRIPTION
**Pull request by andriifromlviv**

### CODE REVIEW

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuComponents/UserMenuHeaderButtonPageComponent.cs
Comment (dmatsev) [inline TAF_EasyRest/TestFramework/PageComponents/NavigationMenuComponents/UserMenuHeaderButtonPageComponent.cs:16]: I think you can use method of base component class

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuComponents/UserMenuHeaderButtonPageComponent.cs
Comment (dmatsev) [inline TAF_EasyRest/TestFramework/PageComponents/NavigationMenuComponents/UserMenuHeaderButtonPageComponent.cs:25]: I think you can use method of base component class

File: TAF_EasyRest/TestFramework/Test/Sample test.cs
Comment (dmatsev) [inline TAF_EasyRest/TestFramework/Test/Sample test.cs:38]: I think you can Sign In steps move to Setup()

File: TAF_EasyRest/TestFramework/Test/Sample test.cs
Comment (dmatsev) [inline TAF_EasyRest/TestFramework/Test/Sample test.cs:40]: admPage - please provide the meaningful name

**Review by dmatsev — DISMISSED** [review]
Approve with comments

**Review by sofiaso — APPROVED** [review]

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuComponents/UserMenuHeaderButtonPageComponent.cs
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/PageComponents/NavigationMenuComponents/UserMenuHeaderButtonPageComponent.cs:16]: agree, I've just not changed this code here

File: TAF_EasyRest/TestFramework/PageComponents/NavigationMenuComponents/UserMenuHeaderButtonPageComponent.cs
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/PageComponents/NavigationMenuComponents/UserMenuHeaderButtonPageComponent.cs:25]: agree, I've just not changed this code here

File: TAF_EasyRest/TestFramework/Test/Sample test.cs
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/Test/Sample test.cs:38]: agree, this test was added only to check waiters. it will be deleted soon.

File: TAF_EasyRest/TestFramework/Test/Sample test.cs
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/Test/Sample test.cs:40]: agree, this test was added only to check waiters. it will be deleted soon.

File: TAF_EasyRest/TestFramework/Test/ClientTests.cs
Comment (LoveMyDaughter) [inline TAF_EasyRest/TestFramework/Test/ClientTests.cs:1]: We must not do this. It will be a big problem later. We have to make changes and build this branch before merge.

**Review by LoveMyDaughter — COMMENTED** [review]
I can't approve without Sofia's agree, because you commented out her code.

File: TAF_EasyRest/TestFramework/PageComponents/BasePageComponent.cs
Comment (taniarezhytska) [inline TAF_EasyRest/TestFramework/PageComponents/BasePageComponent.cs:13]: You can use the `WaitUntilElementIsVisible(By locator, IWebDriver driver, int timeToWait)` method here to not repeat the code.

File: TAF_EasyRest/TestFramework/Test/Sample test.cs
Comment (taniarezhytska) [inline TAF_EasyRest/TestFramework/Test/Sample test.cs:4]: The file name must match the class name.

File: TAF_EasyRest/TestFramework/PageComponents/BasePageComponent.cs
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/PageComponents/BasePageComponent.cs:11]: done

File: TAF_EasyRest/TestFramework/Pages/BasePage.cs
Comment (dmatsev) [inline TAF_EasyRest/TestFramework/Pages/BasePage.cs:32]: I think it is better to make this method an extension method of IWebDriver to avoid code duplication in the BasePage and BasePageComponent classes. What do you think?

**Review by dmatsev — COMMENTED** [review]
I left the comment.

File: TAF_EasyRest/TestFramework/Pages/BasePage.cs
Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/Pages/BasePage.cs]: agree with you, it`s a good idea

Comment (andriifromlviv) [reply TAF_EasyRest/TestFramework/Pages/BasePage.cs]: done

**Review by dmatsev — APPROVED** [review]

**Review by sofiaso — APPROVED** [review]

**Pull request merged**
